src/wallet/db.cpp

```diff
@@ -551,6 +551,7 @@ void BerkeleyBatch::Close()
     {
         LOCK(cs_db);
         --env->mapFileUseCount[strFile];
+        env->m_db_in_use.notify_all();
```

Comment:
Performance nit: notifying while not holding the lock is slightly faster (currently you'll wake the waiting thread, which then immediately needs to sleep again because the lock is still held).

Reply:
Done